python-lib/gluonts_forecasts/utils.py

```diff
@@ -123,3 +123,44 @@ def quantile_forecasts_series(sample_forecasts, quantile, frequency):
     sample_forecasts_quantile = sample_forecasts.quantile(quantile)
     index = pd.date_range(sample_forecasts.start_date, periods=len(sample_forecasts_quantile), freq=frequency)
     return pd.Series(index=index, data=sample_forecasts_quantile)
+
+
+class GPUError(Exception):
+    """Custom exception raised when the GPU selection failed"""
+
+    pass
+
+
+def set_mxnet_context(gpu_devices):
+    """Return the right MXNet context from the selected GPU configuration.
+
+    Args:
+        gpu_devices (list): List of gpu device numbers or 'container_gpu'. Default to None which means no gpu.
+
+    Raises:
+        GPUError:
+            If mx.context.num_gpus() fails on a Cuda error.
+            If no GPUs are detected on the DSS server or in the container.
+
+    Returns:
+        A mxnet.context.Context to use for DL models training.
+    """
+
+    if gpu_devices is None:
```

Comment:
When would that happen?

Reply:
gpu_devices is None when use_gpu is False, so in this case I set the context to mx.context.cpu()

Reply:
Are we sure this will work in the case where the user did the following sequence:
- Activate use_gpu
- Select a GPU device
- Deactivate use_gpu?